### ISSUE TITLE
[a11y]: Focus on the less destructive action on a danger modal

### ISSUE THREAD

**Issue by thefirstartist** [carbon-design-system/carbon#10914]
### Package

carbon-components

### Browser

Chrome

### Operating System

MacOS

### Package version

V11

### React version

V7.54.2

### Automated testing tool and ruleset

n/a

### Assistive technology

n/a

### Description

In a dialog/modal, the default focus should be set to the first focusable element, such as a text field in a form. However, in a danger modal, according to W3 guideline, "Focus is automatically set to the first focusable element inside the dialog, which is the "No" button. This is the least destructive action, so focusing "No" helps prevent users from accidentally confirming the destructive "Discard" action, which cannot be undone."
(https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/alertdialog.html)

### WCAG 2.1 Violation

2.4.3 Focus order

### CodeSandbox example

n/a

### Steps to reproduce

in Carbon's live demo for modal, opening a danger modal shows the default focus is set on the red "Save" button. The focus should be set on the Cancel button instead to prevent users from accidentally confirming the destructive action.

<img width="934" alt="image" src="https://user-images.githubusercontent.com/40870771/156827596-427a988e-4547-4f60-a7a0-ef8feb40dc09.png">

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by joshblack** [comment]
Hi there @thefirstartist! 👋 I believe this should be configurable through the `selectorPrimaryFocus` prop. We can definitely update the demos on the website to reflect this 👍

**Comment by thefirstartist** [comment]
@joshblack Sounds good, Thanks Josh! It would be great if the rule can also be added to the Focus section. :-)
Another question about the modal demo is the button sizes. All the modal examples in the page show full bleed buttons but the live demo shows otherwise. Is it a new rule? I don't see that mentioned in the Sizing section.

**Comment by jnm2377** [comment]
@thefirstartist I think the full bleed discrepancy was a bug and has since been fixed. 

<img width="1026" alt="Screen Shot 2022-08-25 at 12 48 16 AM" src="https://user-images.githubusercontent.com/32556167/186585340-6709611b-3d7f-4d57-9a71-7cd7e04db2bf.png">

I would say the next steps for this would be to add a storybook example with the danger modal having primary focus on the secondary button as discussed here. To take things further, I think we could even explore defaulting to the secondary button if the modal is a danger modal `&&` has a secondary button.
